tests/test_processor.py

```diff
@@ -62,35 +62,37 @@ def test_processor_input_fock_state_with_loss():
 
 @patch.object(pcvl.utils.logging.ExqaliburLogger, "warn")
 def test_processor_input_fock_state_with_all_noise_sources(mock_warn):
-    source = Source(emission_probability=0.2,
-                    multiphoton_component=0.1, multiphoton_model="indistinguishable",
-                    indistinguishability=0.9)
-    source.simplify_distribution = True
-    p = Processor("Naive", Circuit(4), source)
+    nm = NoiseModel(brightness=0.2, indistinguishability=0.9, g2=0.1, g2_distinguishable=False)
+    p = Processor("Naive", Circuit(4), nm)
     with LogChecker(mock_warn):
         p.with_input(BasicState([0, 1, 1, 0]))
 
     expected = {'|0,0,0,0>': 16 / 25,
-                '|0,0,2{_:0},0>': 0.0015490319977879558,
```

Comment:
The noisy mixed state seems to have changed a lot (not the same number of states, for instance). Have you checked the new distribution is OK?

Reply:
I generated it using the release/0.13.0 branch and copied it here (except for the 16/25). 
The problem is that we can't access the `simplify_distribution` attribute of the source anymore (which is not really a problem since it could lead to wrong results in feed-forward simulations with heralded processors as subcircuits), that was set to True in this test.